app-backend/akkautil/src/main/scala/CommonHandlers.scala

```diff
@@ -100,27 +100,55 @@ trait CommonHandlers extends RouteDirectives {
       )
     )
 
-  def authorizeScopeLimitDirective(
+  def isBelowLimitCheck(
       usedLimitFuture: Future[Long],
       domain: Domain,
       action: Action,
       user: User
-  ): Directive0 = {
+  ): Future[Boolean] = {
     val userScopedAction = Scopes.resolveFor(domain, action, user.scope.actions)
     val userLimitOption = userScopedAction.flatMap(_.limit)
 
     // if user does not have an explicit scope limit, assume infinite
-    val isBelowLimit = usedLimitFuture.map { usedLimit =>
+    usedLimitFuture.map { usedLimit =>
       (userScopedAction, userLimitOption) match {
         case (None, _) => false
         case (_, Some(userLimit)) =>
-          usedLimit < userLimit
+          usedLimit <= userLimit
         case _ => true
       }
     }
-    SecurityDirectives.authorizeAsync(isBelowLimit)
   }
 
+  def isBelowLimitCheck(
+      usedLimit: Long,
+      domain: Domain,
+      action: Action,
+      user: User
+  ): Boolean = {
```

Comment:
Is this overload ever used?

Reply:
Yes, it is used here: https://github.com/raster-foundry/raster-foundry/pull/5575/files#diff-717655546bdb9a3b981e85a480d9376bfdaf00a68f21b43ec00ea060a2b7a98cR238 -- `usedLimit` v.s. `usedLimitFuture`

Reply:
Update: discussed this with @jisantuc in a meeting. The conclusion is that this method is being used and it is useful when the limit is not wrapped by a `Future`.